src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -364,10 +365,799 @@ private void DisposeInternal()
+                        }
+                    }
+
+                    return hr;
+                }
+
+                public void Dispose()
+                {
+                    DisposeInternal();
+                    GC.SuppressFinalize(this);
+                }
+
+                private void DisposeInternal()
+                {
+                    if (_disposed)
+                    {
+                        return;
+                    }
+                    _disposed = true;
+                    Marshal.Release(Inst);
+                }
+
+                ~SymUnmanagedScopeRcw()
+                {
+                    DisposeInternal();
+                }
+            }
+        }
+
+        private sealed class SymUnmanagedDocumentWrapperCache : ComWrappers
+        {
+            public static readonly SymUnmanagedDocumentWrapperCache Instance = new SymUnmanagedDocumentWrapperCache();
+
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count) => throw new NotImplementedException();
+            protected override object? CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+            {
+                Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+                return new SymUnmanagedDocumentRcw(externalComObject);
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            public unsafe record SymUnmanagedDocumentRcw(IntPtr Inst) : ISymUnmanagedDocument
+            {
+                private bool _disposed = false;
+
+                public int GetUrl(int bufferLength, out int count, char[] url)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, int*, char*, int>)(*(*(void***)Inst + 3));
+                    int hr;
+
+                    fixed (int* countPtr = &count)
+                    {
+                        if (url == null)
+                        {
+                            hr = func(Inst, bufferLength, countPtr, null);
+                        }
+                        else
+                        {
+                            fixed (char* urlPtr = url)
+                            {
+                                hr = func(Inst, bufferLength, countPtr, urlPtr);
+                            }
+                        }
+                    }
+
+                    return hr;
+                }
+
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                private int SingleByRefGuidWrapper(int methodSolt, ref Guid guid)
+                {
+                    var func = (delegate* unmanaged<IntPtr, Guid*, int>)(*(*(void***)Inst + methodSolt));
+                    int hr;
+
+                    fixed (Guid* guidPtr = &guid)
+                    {
+                        hr = func(Inst, guidPtr);
+                    }
+
+                    return hr;
```

Comment:
```suggestion
                    fixed (Guid* guidPtr = &guid)
                    {
                        return func(Inst, guidPtr);
                    }
```
to match the style of other wrappers.